app/Main.hs

```diff
@@ -108,14 +109,16 @@ data VaultData = VaultData (Map.Map String String)
 instance FromJSON VaultData where
   parseJSON =
     let
+      onlyStrings (Aeson.String t) = Just (unpack t)
+      onlyStrings _ = Nothing
```

Comment:
I went with this approach since as a user of vaultenv I don't really have any control of the format of Vault's responses. It may return something like:

``` JSON
{
  "data": {
    "foo": 42,
    "bar": "foobar"
  }
}
```

and I may only want to use `data.bar`, but since the parser gives up on `data.foo` not being a string I'm effectively entirely blocked.

That said, silently removing it probably isn't the right option. I could see myself thinking that if I use `data.foo`, like this:

```
SOME_VAR=some-secret#foo
```

I would expect `SOME_VAR` to have the string `42`, converting it to a string. However, since it's being silently removed I think you'd get a "Key foo not found for path some-secret", which is probably even more confusing.

I'm leaning towards converting the values to strings, but that leaves the question of what to do with sub-objects.

Another option may be to just parse out the values that we care about from the secrets file (the ones we're actually trying to set in environment variables) instead of trying to parse the entire `data` object, but that would probably require a bit more of a rewrite.

Reply:
Thinking more about it, I think I like your approach more, it seems like the best way to unblock the issue I was having without having to do a larger rewrite.

I was reading through the Vault docs and as far as I can tell Vault doesn't have any restrictions on the `data` object, except that it must have string keys.

Reply:
Ended up pushing another approach that parses to `Aeson.Value` instead and handles non-strings in `lookupSecrets` instead, which looks a lot cleaner to me, and has the benefit that you now get "Key foo in path some-secret is not a String" as an error instead of "Key foo not found for path some-secret" as an error message if you try to use a non-string value in an environment variable.